### PR TITLE
MINOR: AbstractConfigTest.testClassConfigs should reset the class loa…

### PR DESCRIPTION
**Pull request by chia7712**
reference to https://github.com/apache/kafka/pull/10006#issuecomment-770394615

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java [hunk 1]
```diff
@@ -276,52 +277,46 @@ public RestrictedClassLoader() {
...
-        // Properties specified as classes should succeed
-        testConfig = new ClassTestConfig(ClassTestConfig.RESTRICTED_CLASS, Collections.singletonList(ClassTestConfig.RESTRICTED_CLASS));
-        testConfig.checkInstances(ClassTestConfig.RESTRICTED_CLASS, ClassTestConfig.RESTRICTED_CLASS);
-        testConfig = new ClassTestConfig(ClassTestConfig.RESTRICTED_CLASS, Arrays.asList(ClassTestConfig.VISIBLE_CLASS, ClassTestConfig.RESTRICTED_CLASS));
-        testConfig.checkInstances(ClassTestConfig.RESTRICTED_CLASS, ClassTestConfig.VISIBLE_CLASS, ClassTestConfig.RESTRICTED_CLASS);
-        // Properties specified as classNames should fail to load classes
+        ClassLoader origin = Thread.currentThread().getContextClassLoader();
         try {
-            new ClassTestConfig(ClassTestConfig.RESTRICTED_CLASS.getName(), null);
-            fail("Config created with class property that cannot be loaded");
```
Comment (showuon) [inline clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java:308]: It's good refactor to use `assertThrow`, but the failed message is missed. Please add it in `assertThrow`. Thanks.

File: clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java [hunk 2]
```diff
@@ -276,52 +277,46 @@ public RestrictedClassLoader() {
...
-        } catch (ConfigException e) {
-            // Expected Exception
-        }
-        try {
-            testConfig = new ClassTestConfig(null, Arrays.asList(ClassTestConfig.VISIBLE_CLASS.getName(), ClassTestConfig.RESTRICTED_CLASS.getName()));
-            testConfig.getConfiguredInstances("list.prop", MetricsReporter.class);
-            fail("Should have failed to load class");
```
Comment (showuon) [inline clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java:315]: same here

File: clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java [hunk 3]
```diff
@@ -276,52 +277,46 @@ public RestrictedClassLoader() {
...
-        } catch (KafkaException e) {
-            // Expected Exception
-        }
-        try {
-            testConfig = new ClassTestConfig(null, ClassTestConfig.VISIBLE_CLASS.getName() + "," + ClassTestConfig.RESTRICTED_CLASS.getName());
-            testConfig.getConfiguredInstances("list.prop", MetricsReporter.class);
-            fail("Should have failed to load class");
```
Comment (showuon) [inline clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java:322]: same here

**Review by showuon — COMMENTED** [review]
@chia7712 , Thanks for the fix. Left some minor comments.

**Comment by chia7712** [comment]
@showuon Thanks for your reviews. I have addressed your comment by https://github.com/apache/kafka/pull/10010/commits/dac031f228c8e8db02de2f17e14f43a96db11b8e

**Review by showuon — APPROVED** [review]
LGTM

File: clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java [hunk 4]
```diff
@@ -276,52 +277,49 @@ public RestrictedClassLoader() {
...
-
-        // Test class overrides where some classes are not visible to thread context classloader
-        Thread.currentThread().setContextClassLoader(restrictedClassLoader);
-        // Properties specified as classes should succeed
-        testConfig = new ClassTestConfig(ClassTestConfig.RESTRICTED_CLASS, Collections.singletonList(ClassTestConfig.RESTRICTED_CLASS));
-        testConfig.checkInstances(ClassTestConfig.RESTRICTED_CLASS, ClassTestConfig.RESTRICTED_CLASS);
-        testConfig = new ClassTestConfig(ClassTestConfig.RESTRICTED_CLASS, Arrays.asList(ClassTestConfig.VISIBLE_CLASS, ClassTestConfig.RESTRICTED_CLASS));
-        testConfig.checkInstances(ClassTestConfig.RESTRICTED_CLASS, ClassTestConfig.VISIBLE_CLASS, ClassTestConfig.RESTRICTED_CLASS);
-        // Properties specified as classNames should fail to load classes
+        ClassLoader origin = Thread.currentThread().getContextClassLoader();
```
Comment (hachikuji) [inline clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java]: nit: maybe `originalClassLoader` to be consistent with other names?

**Review by hachikuji — APPROVED** [review]
LGTM. Left a minor naming suggestion.

**Pull request merged**